### ISSUE TITLE
x86 paging

### ISSUE THREAD

**Issue by SamTebbs33** [SamTebbs33/pluto#7]
Paging needs to be implemented to map the higher half kernel to physical memory. This will also require a simple watermark memory manager that takes the end of the loaded kernel code and allocates memory on a linear basis after that (this may require a separate ticket).

4KB paging will be looked at later on.

Depends on #6, #22 and #19

**Comment by SamTebbs33** [comment]
Currently blocked on https://github.com/ziglang/zig/issues/2616

**Comment by SamTebbs33** [comment]
Having some trouble with triple faults when using 4KB paging. 4MB paging works fine.

**Comment by SamTebbs33** [comment]
I'll rework this issue for 4MB and get back to 4KB paging later on.
